File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
x = 1
num_list = []

def d(x):
    s = str(x)
    num = x
    for i in s:
        num += int(i)
    return num


while x <= 10000:
    try:
        num_list.append(d(x))
        x+=1
    except:
        break
        
for i in range(1, 10001):
    if i not in num_list:
        print(i)

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.cc
#include <iostream>

bool ifselfnum[20000]={false};

int selfnum(int n){
    int nsum=n;
    while(n != 0){ // 모든 자리수를 다 더할 때까지
        nsum += n%10; //일의 자리
        n = n/10;
    }
    return nsum;
}

int main(){
    int num;
    for (int i=1; i<10001; i++){ //
        num = i;
        while(num<10001){
            num = selfnum(num);
            if ((ifselfnum[num]==false)&&(num<10001)){
                ifselfnum[num]=true;
            }
        }
    }

    for (int j=1; j<10001; j++){
        if (ifselfnum[j]==false){
            std::cout << j << '\n';
        }
    }
}